src/components/ManageCDP.jsx

```diff
@@ -0,0 +1,272 @@
+import React, { useState } from "react";
+import { useDispatch } from "react-redux";
+import { useNavigate } from "react-router-dom";
+import styled, {css} from "styled-components";
+import Effect from "./Effect";
+import ToolTip from "./ToolTip";
+import PrimaryButton from "./PrimaryButton";
+import { addCollateral, mint } from "../transactions/cdp";
+import { handleTxError } from "../wallets/wallets";
+import { setAlert } from "../redux/slices/alertSlice";
+import LoadingOverlay from "./LoadingOverlay";
+
+export default function ManageCDP({cdp, price, setCurrentCDP}){
+    const [supplyInput, setSupplyInput] = useState(0);
+    const [supplyPrice, setSupplyPrice] = useState(0);
+    const [borrowInput, setBorrowInput] = useState(0);
+    const [borrowPrice, setBorrowPrice] = useState(0);
+    const [modalVisible, setModalVisible] = useState(false);
+    const [modalCanAnimate, setModalCanAnimate] = useState(false);
+    const [loading, setLoading] = useState(false);
+    const [loadingText, setLoadingText] = useState(null);
+    const navigate = useNavigate();
+    const dispatch = useDispatch();
+
+    const [additionalSupply, setAdditionalSupply] = useState("")
+    const [additionalBorrow, setAdditionalBorrow] = useState("")
+
+    const handleAddSupply = (event) => {
+        setAdditionalSupply(event.target.value === "" ? "" : Number(event.target.value));
+      };
+
+    const handleAddBorrow = (event) => {
+    setAdditionalBorrow(event.target.value === "" ? "" : Number(event.target.value));
+    };
+
+    var supplyDetails = [
+        {
+            title: "Borrow Limit",
+            val: `${Math.max(
+                0,
+                Math.trunc(
+                  (100 * ((price * cdp.collateral) / 1000000)) /
+                    1.4 -
+                    (100 * cdp.debt) / 1000000,
+                ) / 100,
+              )}`,
+            hasToolTip: true,
+        },
+        {
+            title: "Supply APY",
```

Comment:
We should use APR to be consistent. This value will probably be 0 unless the supplied thing is Algo's right?

Reply:
Make sense.